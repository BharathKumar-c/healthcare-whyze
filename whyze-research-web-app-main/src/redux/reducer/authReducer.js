import { createSlice } from '@reduxjs/toolkit';
import jwt_decode from 'jwt-decode';

const initialState = {
  isLoggedIn: loginCheck(),
  accessToken: null,
  refreshToken: null,
  isLoading: false,
  isDisableForgot: false,
  message: '',
  userDetails: {},
  profileCount: 0,
};

function loginCheck() {
  const credentials = JSON.parse(localStorage.getItem('credentials'));
  if (credentials?.access_token) {
    const { exp } = jwt_decode(credentials.access_token);
    if (Date.now() <= exp * 1000) {
      return true;
    }
  }
  return false;
}

const authSlice = createSlice({
  name: 'authAction',
  initialState: initialState,
  reducers: {
    setLoginStatus: (state, { payload }) => {
      state.isLoggedIn = payload;
    },
    setAccessToken: (state, { payload }) => {
      state.accessToken = payload;
    },
    setRefreshToken: (state, { payload }) => {
      state.refreshToken = payload;
    },
    setLoading: (state, { payload }) => {
      state.isLoading = payload;
    },
    setDisableForgot: (state, { payload }) => {
      state.isDisableForgot = payload;
    },
    setResetMessage: (state, { payload }) => {
      state.message = payload;
    },
    setUserDetails: (state, { payload }) => {
      state.userDetails = payload;
    },
    setProfileCount: (state) => {
      state.profileCount = state.profileCount + 1;
    },
  },
});

export const {
  setLoginStatus,
  setAccessToken,
  setRefreshToken,
  setLoading,
  setDisableForgot,
  setResetMessage,
  setUserDetails,
  setProfileCount,
} = authSlice.actions;

export default authSlice.reducer;
