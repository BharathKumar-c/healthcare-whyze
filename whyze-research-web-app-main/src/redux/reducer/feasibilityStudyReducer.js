import { createSlice } from '@reduxjs/toolkit';
import { v4 as uuid } from 'uuid';

export const initialState = {
  segments: [
    {
      name: 'Segment-1',
      isDone: false,
      activeTab: {
        name: 'feasibility',
        index: 0,
      },
      tabs: [
        {
          name: 'feasibility',
          title: 'Feasibility',
          isActive: true,
          position: 'left',
        },
      ],
      feasibilityStudy: [
        {
          name: 'Indication',
          case: 'Indication',
          condition: 'Inclusion',
          caseName: '',
          child: [],
        },
        {
          name: 'Demographics',
          case: 'Demographics',
          condition: 'Inclusion',
          caseName: '',
          child: [
            {
              id: uuid().slice(0, 8),
              condition: 'Gender',
              field: 'gender',
              gender: '',
            },
            {
              id: uuid().slice(0, 8),
              condition: 'Age range',
              field: 'age',
              ageRange: {
                min: 0,
                max: 0,
              },
            },
          ],
        },
      ],
      country: [],
      location: [],
      selectedCountries: [],
      totalConnectedNotConnected: {},
      sites: [],
      sitesContacted: [],
      sitesInitiated: [],
      preferredSites: [],
      patientCount: {
        feasibility: 0,
        location: 0,
      },
      siteCount: {
        feasibility: 0,
        location: 0,
      },
    },
  ],
  cardValue: {
    case: 'Indication',
    condition: '',
    index: 0,
  },
  activeSegment: {
    name: '',
    index: 0,
  },
};

const feasibilityStudySlice = createSlice({
  name: 'feasibilityStudyAction',
  initialState: initialState,
  reducers: {
    setCardDetails: (state, { payload }) => {
      state.cardValue = payload;
    },
    setSegments: (state, { payload }) => {
      state.segments = payload;
    },
    setActiveSegment: (state, { payload }) => {
      state.activeSegment = payload;
    },
    resetSegments: (state) => {
      state.segments = initialState.segments;
      state.cardValue = {
        case: 'Indication',
        condition: '',
        index: 0,
      };
      state.activeSegment = {
        name: '',
        index: 0,
      };
    },
  },
});

export const { setCardDetails, setSegments, setActiveSegment, resetSegments } =
  feasibilityStudySlice.actions;

export default feasibilityStudySlice.reducer;
