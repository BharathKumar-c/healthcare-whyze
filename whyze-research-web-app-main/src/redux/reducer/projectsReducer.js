import { createSlice } from '@reduxjs/toolkit';

const initialState = {
  projects: [],
  projectId: '',
  clinicalTrial: '',
  sponsor: '',
  description: '',
  measurement: 'Metric',
  standards: '',
  segmentId: [],
};

const projectSlice = createSlice({
  name: 'projectAction',
  initialState: initialState,
  reducers: {
    setProjects: (state, { payload }) => {
      state.projects = payload;
    },
    setProjectId: (state, { payload }) => {
      state.projectId = payload;
    },
    setClinicalTrial: (state, { payload }) => {
      state.clinicalTrial = payload;
    },
    setSponsor: (state, { payload }) => {
      state.sponsor = payload;
    },
    setDescription: (state, { payload }) => {
      state.description = payload;
    },
    setStandards: (state, { payload }) => {
      state.standards = payload;
    },
    setMeasurement: (state, { payload }) => {
      state.measurement = payload;
    },
    setSegmentId: (state, { payload }) => {
      state.segmentId = payload;
    },
    setProjectSettings: (state, { payload }) => {
      state.projectId = payload.projectId;
      state.segmentId = payload.segmentId;
      state.clinicalTrial = payload.clinicalTrial;
      state.sponsor = payload.sponsor;
      state.description = payload.description;
      state.measurement = payload.measurement;
      state.standards = payload.standards;
    },
    clearSettings: (state) => {
      state.clinicalTrial = '';
      state.sponsor = '';
      state.description = '';
      state.measurement = 'Metric';
      state.standards = '';
      state.projectId = '';
      state.segmentId = [];
    },
  },
});

export const {
  setProjects,
  setProjectId,
  setProjectSettings,
  setClinicalTrial,
  setDescription,
  setSponsor,
  setMeasurement,
  setStandards,
  setSegmentId,
  clearSettings,
} = projectSlice.actions;

export default projectSlice.reducer;
