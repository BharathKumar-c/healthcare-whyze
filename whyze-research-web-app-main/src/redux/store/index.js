import { configureStore } from '@reduxjs/toolkit';
import reducer from '../reducer/combinedReducer';

const store = configureStore({
  reducer: reducer,

  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});

export default store;
