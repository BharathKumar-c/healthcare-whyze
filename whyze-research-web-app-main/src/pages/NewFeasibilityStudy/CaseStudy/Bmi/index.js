import React from 'react';
import PropTypes from 'prop-types';
import { Select, Text } from '../../../../components/basic';
import { bmiRangeOptions } from '../../../../constant';
import { ConditionContainer } from '../../../../components/shared';
import { Row, Col, Tag } from 'antd';
import { Drag, Vector } from '../../../../assets';
import { useSelector } from 'react-redux';
import { useDispatch } from 'react-redux';
import { setSegments } from '../../../../redux/reducer/feasibilityStudyReducer';
import { totalPatientFormatter } from '../../../../utils/calculations';
import useNewFeasibilityStudyService from '../../useNewFeasibilityStudyService';

const Bmi = (props) => {
  const { segments, activeSegment } = useSelector(
    (state) => state.feasibilityStudyReducer,
  );
  const dispatch = useDispatch();
  const { optimusValue } = useNewFeasibilityStudyService();

  const handleSelect = (range) => {
    const study = JSON.parse(JSON.stringify(segments));
    study[activeSegment.index].feasibilityStudy[
      props.parentIndex
    ].child[0].bmiRange = range;

    optimusValue({
      id: props.caseDetails.child[0].id,
      data: study[activeSegment.index]?.feasibilityStudy,
    });

    dispatch(setSegments(study));
  };

  return (
    <div className="bmi_container">
      <ConditionContainer className={'condition-container'} isActive isVertical>
        <Row gutter={{ sm: 32, md: 32, lg: 32 }} className="bmi-wrapper">
          <Col className="bmi_drag-icon">
            <Drag />
          </Col>
          <Col>
            <Text className="bmi-wrapper-title" text="BMI Range" />
          </Col>
          <Col>
            <div
              className="condition-container-select"
              id="condition-container-select"
            >
              <Select
                lable="Select type"
                options={bmiRangeOptions}
                getPopupContainer={() =>
                  document.getElementById('condition-container-select')
                }
                name="range"
                value={props.caseDetails?.child[0]?.bmiRange}
                handleSelect={handleSelect}
              />
            </div>
          </Col>
        </Row>
        <Row className="indication-no-patient-tag-row">
          <Tag className="indication-no-patient-tag">
            <Vector />
            {totalPatientFormatter(props.caseDetails?.child[0]?.count || 0)}
          </Tag>
        </Row>
      </ConditionContainer>
    </div>
  );
};

Bmi.propTypes = {
  caseDetails: PropTypes.object,
  parentIndex: PropTypes.number,
};

Bmi.defaultProps = {
  caseDetails: null,
  parentIndex: 0,
};

export default Bmi;
