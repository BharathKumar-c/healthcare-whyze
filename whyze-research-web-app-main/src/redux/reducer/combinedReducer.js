import { combineReducers } from 'redux';
import feasibilityStudy from './feasibilityStudyReducer';
import project from './projectsReducer';
import auth from './authReducer';

export default combineReducers({
  feasibilityStudyReducer: feasibilityStudy,
  authReducer: auth,
  projectReducer: project,
});
