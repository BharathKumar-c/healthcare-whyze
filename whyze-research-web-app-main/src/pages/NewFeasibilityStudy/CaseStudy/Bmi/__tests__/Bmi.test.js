import React from 'react';
import { act, fireEvent, render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { setSegments } from '../../../../../redux/reducer/feasibilityStudyReducer';
import store from '../../../../../redux/store';
import Bmi from '../index';

function renderWithContext(element) {
  const { container, getByText } = render(
    <Provider store={store}>{element}</Provider>,
  );
  return { container, store, getByText };
}
const caseDetails = {
  name: 'Bmi',
  case: 'Bmi',
  condition: 'Inclusion',
  caseName: '',
  child: [],
};

describe('Test cases for BMI', () => {
  test('Create the snapshots for BMI', () => {
    const { container } = renderWithContext(<Bmi />);

    act(() => {
      const { segments, activeSegment } = JSON.parse(
        JSON.stringify(store.getState().feasibilityStudyReducer),
      );
      segments[activeSegment.index].feasibilityStudy.push(caseDetails);
      store.dispatch(setSegments(segments));
    });

    expect(container).toMatchSnapshot();
  });

  test('Render Components in BMI', () => {
    renderWithContext(<Bmi />);
    const selectBox = screen.getByRole('combobox');
    expect(selectBox).toBeInTheDocument();
    fireEvent.change(selectBox, {
      target: { value: 'obesity' },
    });
    const contentText = screen.getByText('Obesity (>=30)');
    expect(contentText).toBeInTheDocument();
  });
});
