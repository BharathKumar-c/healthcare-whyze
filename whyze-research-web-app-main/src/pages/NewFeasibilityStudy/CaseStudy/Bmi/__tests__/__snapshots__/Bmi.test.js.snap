// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test cases for BMI Create the snapshots for BMI 1`] = `
<div>
  <div
    class="bmi_container"
  >
    <div
      class="ant-space ant-space-horizontal ant-space-align-center condition-container"
      style="border: 2px solid #d3d6d9; display: flex; flex-direction: column; margin-bottom: 0.5rem; padding: 20px 30px 12px 20px;"
    >
      <div
        class="ant-space-item"
      >
        <div
          class="ant-row bmi-wrapper"
          style="margin-left: -16px; margin-right: -16px;"
        >
          <div
            class="ant-col bmi_drag-icon"
            style="padding-left: 16px; padding-right: 16px;"
          >
            <svg>
              Drag.svg
            </svg>
          </div>
          <div
            class="ant-col"
            style="padding-left: 16px; padding-right: 16px;"
          >
            <span
              class="ant-typography bmi-wrapper-title"
            >
              BMI Range
            </span>
          </div>
          <div
            class="ant-col"
            style="padding-left: 16px; padding-right: 16px;"
          >
            <div
              class="condition-container-select"
            >
              <div
                class="alignment"
              >
                <span
                  class="ant-typography select-lable"
                >
                  Select type
                </span>
                <div
                  class="ant-select ant-select-single ant-select-show-arrow"
                  name="range"
                >
                  <div
                    class="ant-select-selector"
                  >
                    <span
                      class="ant-select-selection-search"
                    >
                      <input
                        aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                        aria-autocomplete="list"
                        aria-controls="rc_select_TEST_OR_SSR_list"
                        aria-haspopup="listbox"
                        aria-owns="rc_select_TEST_OR_SSR_list"
                        autocomplete="off"
                        class="ant-select-selection-search-input"
                        id="rc_select_TEST_OR_SSR"
                        readonly=""
                        role="combobox"
                        style="opacity: 0;"
                        type="search"
                        unselectable="on"
                        value=""
                      />
                    </span>
                    <span
                      class="ant-select-selection-item"
                      title="Underweight (<18.5)"
                    >
                      Underweight (&lt;18.5)
                    </span>
                  </div>
                  <span
                    aria-hidden="true"
                    class="ant-select-arrow"
                    style="user-select: none;"
                    unselectable="on"
                  >
                    <span
                      aria-label="down"
                      class="anticon anticon-down ant-select-suffix"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="down"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                        />
                      </svg>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
